src/storage/mod.rs

```diff
@@ -292,3 +305,28 @@ fn should_detect_zeroes() {
   let nums = vec![1; 10];
   assert!(not_zeroes(&nums));
 }
+
+#[cfg(test)]
+mod test {
+  use super::*;
+  use crypto::generate_keypair;
+  use sha2::Sha512;
+  use ed25519_dalek::{Keypair};
+
+  #[test]
+  fn should_write_and_read_keypair() {
+    let keypair = generate_keypair();
+    let mut storage = Storage::new_memory().unwrap();
+    assert!(storage.write_keypair(&keypair).is_ok(), "Keypair written successfully");
+
+    let read = storage.read_keypair();
+    assert!(read.is_ok(), "Keypair readen successfully");
+    let readen_keypair: Keypair = read.unwrap();
+
+    let message: &[u8] = "Hello world".as_bytes();
+    let sig: Signature = keypair.sign::<Sha512>(&message);
+
+    assert!(readen_keypair.verify::<Sha512>(&message, &sig),
```

Comment:
I thought that it'd be a good idea to `verify` with a signature from the original keypair. What's weird is that rust tells me that this is a `bool` although a test is done with `is_ok` [in the ed25519 library](https://github.com/dalek-cryptography/ed25519-dalek/blob/d472254ad3805554792d3be09d94fb3650b05660/src/ed25519.rs#L1409).

I also thought about comparing the two array of bytes (`[u8; KEYPAIR_LENGTH]`) but I haven't found any elegant way of doing it!